test/unittests/util/test_util.py

```diff
@@ -0,0 +1,226 @@
+        space.percent = 95.0
+        space.free = 2 * 1024 * 1024  # 2MB
+        curate_cache(cache_dir)
+        self.assertTrue(exists(aldous_path))
+        self.assertTrue(exists(huxley_path))
+
+        # Free space and percentage low
+        space.percent = 96.0
+        space.free = 2 * 1024 * 1024  # 2MB
+        curate_cache(cache_dir)
+        self.assertFalse(exists(aldous_path))
+        self.assertFalse(exists(huxley_path))
+
+
+@mock.patch('mycroft.configuration.Configuration')
+@mock.patch('mycroft.util.audio_utils.subprocess')
+class TestPlaySounds(TestCase):
+    def test_play_ogg(self, mock_subprocess, mock_conf):
+        mock_conf.get.return_value = test_config
+        play_ogg('insult.ogg')
+        mock_subprocess.Popen.assert_called_once_with(['mock_ogg',
+                                                       'insult.ogg'],
+                                                      env=Anything())
+
+    def test_play_mp3(self, mock_subprocess, mock_conf):
+        mock_conf.get.return_value = test_config
+        play_mp3('praise.mp3')
+        mock_subprocess.Popen.assert_called_once_with(['mock_mp3',
+                                                       'praise.mp3'],
+                                                      env=Anything())
+
+    def test_play_wav(self, mock_subprocess, mock_conf):
+        mock_conf.get.return_value = test_config
+        play_wav('indifference.wav')
+        mock_subprocess.Popen.assert_called_once_with(['mock_wav',
+                                                       'indifference.wav'],
+                                                      env=Anything())
+
+    def test_play_audio_file(self, mock_subprocess, mock_conf):
+        mock_conf.get.return_value = test_config
+        play_audio_file('indifference.wav')
+        mock_subprocess.Popen.assert_called_once_with(['mock_wav',
+                                                       'indifference.wav'],
+                                                      env=Anything())
+        mock_subprocess.Popen.reset_mock()
+
+        play_audio_file('praise.mp3')
+        mock_subprocess.Popen.assert_called_once_with(['mock_mp3',
+                                                       'praise.mp3'],
+                                                      env=Anything())
+        mock_subprocess.Popen.reset_mock()
+        mock_conf.get.return_value = test_config
+        play_audio_file('insult.ogg')
+        mock_subprocess.Popen.assert_called_once_with(['mock_ogg',
+                                                       'insult.ogg'],
+                                                      env=Anything())
+
+
+class TestStringFunctions(TestCase):
+    def test_camel_case_split(self):
+        """Check that camel case string is split properly."""
+        self.assertEqual(camel_case_split('MyCoolSkill'), 'My Cool Skill')
+        self.assertEqual(camel_case_split('MyCOOLSkill'), 'My COOL Skill')
+
+    def test_get_http(self):
```

Comment:
If you wanted to go comprehensive, we could add one to check that only the protocol is modified, eg:
```
self.assertEqual(get_http('http://https.com/'), 'http://https.com/')
```

Reply:
The `get_http('https://github.com/https')` one sort of does that test, but I'll add the one with server url as well.